### PR TITLE
feat: add paid and unpaid executive course variants

### PR DESCRIPTION
**Pull request by uzairr**
## Description:
This PR will add executive education variants for paid and unpaid modes.

For better understanding of executive education please visit this [Confluence page](https://openedx.atlassian.net/wiki/spaces/IM/pages/3333161057/Executive+Education+Course+Type)

## Related PRs:
1: [ecommerce](https://github.com/openedx/ecommerce/pull/3669)
2: [edx-platform](https://github.com/openedx/edx-platform/pull/29999)

## Linked Ticket:
for more details please visit: [PROD-2708](https://openedx.atlassian.net/browse/PROD-2708)

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -2024,7 +2024,11 @@ class Seat(DraftModelMixin, TimeStampedModel):
     CREDIT = 'credit'
     MASTERS = 'masters'
     EXECUTIVE_EDUCATION = 'executive-education'
-    ENTITLEMENT_MODES = [VERIFIED, PROFESSIONAL, EXECUTIVE_EDUCATION]
+    PAID_EXECUTIVE_EDUCATION = 'paid-executive-education'
+    UNPAID_EXECUTIVE_EDUCATION = 'unpaid-executive-education'
+    ENTITLEMENT_MODES = [
+        VERIFIED, PROFESSIONAL, EXECUTIVE_EDUCATION, PAID_EXECUTIVE_EDUCATION, UNPAID_EXECUTIVE_EDUCATION
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: unpaid mode is not an entitlement mode

**Review by DawoudSheraz — CHANGES_REQUESTED** [review]
added comment to address

**Review by DawoudSheraz — APPROVED** [review]
Please add the link to Enrollment track Confluence page in the PR description so the reviewers can better understand the usage of this addition.

**Review by azanbinzahid — APPROVED** [review]

**Review by waheedahmed — COMMENTED** [review]
What is the difference between `EXECUTIVE_EDUCATION` and these new types? Is this just a rename and are we going to remove the old one?

**Comment by uzairr** [comment]
@waheedahmed `EXECUTIVE_EDUCATION` mode was already part of the edX ecosystem but it was not been used for a longer time. The new types are the separate product lines which are created to import 2U courses and has nothing to do with it.

**Review by waheedahmed — APPROVED** [review]

**Comment by DawoudSheraz** [comment]
Merging the PR because the test failure on mysql80 is a known flaky test. mysql80 is not a required check either. Re-running takes about 1 hours minimum which is wasting a lot of time.

**Pull request merged**